xarray_beam/_src/core.py

```diff
@@ -304,3 +304,32 @@ def expand(self, pcoll):
             self._key_to_chunks, num_threads=self.num_threads
         )
     )
+
+
+def validate_chunk(key: Key, dataset: xarray.Dataset) -> None:
+  """Verify that keys correpond to Dataset properties."""
+  missing_keys = [f"'{k}'" for k in key.offsets.keys() if k not in dataset.dims]
+  if missing_keys:
+    raise ValueError(
+      f"Key offset(s) {', '.join(missing_keys)} not found in Dataset dimensions."
+    )
+
+  if key.vars is None:
+    return
+  missing_vars = [f"'{v}'" for v in key.vars if v not in dataset.data_vars]
+  if missing_vars:
+    raise ValueError(
+      f"Key var(s) {', '.join(missing_keys)} not found in Dataset data variables."
+    )
+
+
+class ValidateChunk(beam.PTransform):
```

Comment:
I wonder if this is the best name -- does it make it obvious that it only deals with a single chunk at a time?

In principle one could also check an entire PCollection for consistent but it seems like the overhead would be prohibitive.

Reply:
I think the name makes sense – would updating the docstring suffice? 
Otherwise, naming is hard: any ideas? What about `ValidateEachChunk`?  Or, `ChunksMatchDatasets`

Maybe I can write a follow up CL that adds a combiner to check that the entire collection is valid. We could do this in some sort of streaming fashion for better performance. Also, if we introduced this combiner, it could make the names clearer  (`ValidateChunk` vs `ValidateChunks` or `ValidateEachChunk` vs `ValidateAllChunks`). 

What do you think?

Reply:
I like `ValidateEachChunk`!